crates/wit/src/commands/pull.rs

```
@@ -0,0 +1,273 @@
+                        let release_version = &release.version;
+                        if wit_version != release_version {
+                            terminal.warn(format!("Release version {release_version} doesn't match WIT package version {wit_version}"))?;
+                        }
+                    }
+
+                    for (package_id, package) in &decoded.resolve().packages {
+                        let name = &package.name;
+                        if !existing_pkgs.insert(name.clone()) {
+                            continue;
+                        }
+                        let path = self.write_package(decoded.resolve(), package_id)?;
+                        terminal.status(
+                            "Wrote",
+                            format!("package {name} to '{path}'", path = path.display()),
+                        )?;
+                    }
+                }
+                None => {
+                    terminal.warn(format!("No package found for {pkg}"))?;
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn parse_packages_state(&self) -> Result<(BTreeSet<PackageName>, BTreeSet<PackageName>)> {
+        let mut resolve = Resolve::new();
+
+        // If Resolve::push_dir succeeds there are no unresolved deps
+        match resolve.push_dir(&self.wit_dir) {
+            Ok(_) => {
+                return Ok((
+                    resolve.package_names.into_keys().collect(),
+                    Default::default(),
+                ))
+            }
+            Err(err) => log::debug!("Couldn't resolve packages: {err:#}"),
+        }
+
+        // Approximate the Resolve::push_dir process
+        // TODO: Try to expose this logic from wit-parser instead
+        let mut pkgs = match UnresolvedPackage::parse_dir(&self.wit_dir) {
+            Ok(pkg) => BTreeSet::from([pkg.name]),
+            Err(err) => {
+                log::debug!("Couldn't parse root package: {err:?}");
```

This seems like something we shouldn't silently ignore - but either error on or emit a warning.

---

I added a warning if the root package is unparsable and you didn't pass any explicit packages.

I think it would be somewhat common to have an unparsable root exactly when you are running this command, e.g. bootstrapping a new project or in the middle of making changes that need new dependencies.